addons/analytic/models/analytic_plan.py

```diff
@@ -244,38 +243,47 @@ def _get_applicability(self, **kwargs):
 
     def unlink(self):
         # Remove the dynamic field created with the plan (see `_inverse_name`)
+        if not self:
+            return super().unlink()
```

Comment:
This can be removed now that `_find_plan_column` doesn't require exactly one record anymore